x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SearchableSnapshotAction.java

```diff
@@ -116,7 +116,6 @@ public String getSnapshotRepository() {
         // here before generating snapshots that can't be used if the user doesn't have the right license level.
         BranchingStep conditionalSkipActionStep = new BranchingStep(preActionBranchingKey, checkNoWriteIndex, nextStepKey,
             (index, clusterState) -> {
-                XPackLicenseState licenseState = XPackPlugin.getSharedLicenseState();
```

Comment:
Can you explain why we need to move the license state back up into the top level instead of getting it here? It added quite a few lines injecting it into every ILM action

Reply:
The problem is that `XPackPlugin.getSharedLicenseState()` is backed by a static `SetOnce` field making it unsuitable for use in tests. This pattern of passing in the `licenseState` was taken from ML. In CCR and Autoscaling, the `licenceState` is wrapped in a checker that is passed around instead. For ILM, the ML approach felt most natural since it allows any license check in the future.

I think there is no internal cluster tests today that trigger the `SearchableSnapshotAction`, however, I needed to add one in `FrozenExistenceDeciderIT`, necessitating this change.

If there is a different and/or better pattern for doing this I should be more than happy to adapt this.